test/e2e/bundle_e2e_test.go

```diff
@@ -165,6 +165,142 @@ var _ = Describe("Installing bundles with new object types", func() {
 			}).Should(Succeed())
 		})
 	})
+
+	When("A bundle is installed with a CR that is associated with a provided CRD", func() {
+		By("including the VPA CRD in the CSV")
+		const (
+			packageName = "busybox"
+			channelName = "alpha"
+			subName     = "test-subscription"
+			vpaName     = "busybox-vpa"
+			vpaGroup    = "autoscaling.k8s.io"
+			vpaVersion  = "v1"
+			vpaResource = "verticalpodautoscalers"
+		)
+
+		BeforeEach(func() {
+			const (
+				sourceName = "test-catalog"
+				imageName  = "quay.io/olmtest/single-bundle-index:vpa-race"
+			)
+			var installPlanRef string
+			// create catalog source
+			source := &v1alpha1.CatalogSource{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       v1alpha1.CatalogSourceKind,
+					APIVersion: v1alpha1.CatalogSourceCRDAPIVersion,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      sourceName,
+					Namespace: testNamespace,
+					Labels:    map[string]string{"olm.catalogSource": sourceName},
+				},
+				Spec: v1alpha1.CatalogSourceSpec{
+					SourceType: v1alpha1.SourceTypeGrpc,
+					Image:      imageName,
+				},
+			}
+
+			source, err := operatorClient.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Create(context.TODO(), source, metav1.CreateOptions{})
```

Comment:
nit: I recommend using the [controller-runtime client](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/test/e2e/ctx/ctx.go#L78), since it's less verbose.